wagtail/core/models.py

```diff
@@ -1943,6 +1943,14 @@ class BaseCollectionManager(models.Manager):
     def get_queryset(self):
         return TreeQuerySet(self.model).order_by('path')
 
+    def for_display(self):
+        # This will output the Root collection first, and alphabetize the rest
+        return TreeQuerySet(self.model).annotate(
+            display_order=Case(
+                When(id=get_root_collection_id(), then=Value('-1')),
```

Comment:
`When(depth=1, then=Value('-1'))` would save a query here.

Actually, isn't this relying on the non-root collections having names that sort after the string '-1'? I'm not sure if that's something we can necessarily count on - `then=Value('')` is probably more reliable.